internal/shim-sgx/src/handler/enarx.rs

```diff
@@ -55,37 +52,27 @@ impl<'a> EnarxSyscallHandler for super::Handler<'a> {
 
         // Retrieve TargetInfo from sallyport block and call EREPORT to
         // create Report from TargetInfo.
-        let mut ti = [0u8; 512];
-        let ti_len = ti.len();
+        let mut ti_buf = [0u8; 512];
```

Comment:
I'm looking at this code:
```
        // Generate Report
        let mut target_info: TargetInfo = Default::default();
        let mut f = [0u8; 8];
        let mut x = [0u8; 8];
        f.copy_from_slice(&ti_buf[32..40]);
        x.copy_from_slice(&ti_buf[40..48]);
        let f = u64::from_le_bytes(f);
        let x = u64::from_le_bytes(x);
        target_info.mrenclave.copy_from_slice(&ti_buf[0..32]);
        target_info.attributes = [f, x];
```

@haraldh why cannot just take a direct copy of `ti_buf`?

Reply:
I have no idea... never reviewed that code.. I know that is doing that, because of wanting to fill out the structure in a non-unsafe way and the `attributes` are `packed` and mis-aligned u64.

Reply:
```
#[repr(C, packed(4))]
pub struct Attributes {
...
```

Reply:
needed for:
```
#[repr(C)]
pub struct Measurement {
    misc: Masked<MiscSelect>,
    reserved0: [u8; 20],
    attr: Masked<Attributes>,
    mrenclave: [u8; 32],
    reserved1: [u8; 32],
    isv_prod_id: isv::ProdId,
    isv_svn: isv::Svn,
}
```

Reply:
If `Measurement` is not needed, one could remove the `packed` attribute from the `.attributes` field

Reply:
What do you think:
```rust
        let target_info =
            unsafe {
                let mut target_info_buf = [0u8; 512];
                c.copy_into_slice(512, &mut target_info_buf[..]).or(Err(libc::EFAULT))?;
                transmute::<[u8; 512], TargetInfo>(target_info_buf)
            };
```

Reply:
I'll go with this for the next version.